### PR TITLE
`Development:` Add JavaDocs to UtilServices and Factories [4]

### PR DESCRIPTION
**Pull request by laadvo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper functions

### Description

Added documentation to following files:
- [x] exercise/ExerciseFactory.java
- [x] exercise/ExerciseUtilService.java

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### Test Coverage

unchanged

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java [hunk 1]
```diff
@@ -94,13 +94,25 @@ public class ExerciseUtilService {
...
         exercise.setBonusPoints(10.0);
         return exerciseRepo.save(exercise);
     }
 
+    /**
+     * Adds grading instructions from three different grading criteria to an exercise. One criterion is empty.
+     *
+     * @param exercise The exercise to which the grading instructions should be added.
+     * @return The list of grading criteria.
+     */
```
Comment (jakubriegel) [inline src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java:115]: Should we write here more about what kind of instructions are added? Like `default` or `test`?

**Review by b-fein — APPROVED** [review]

File: src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java (hunk 1 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java:115]: I'm not exactly sure what you mean ba `default` and `test` instructions. If you mean the variable names `testCriterion` and `testCriterion2`, I looked into it and saw that this is only based on the title of the corresponding criterion. I don't think we actually have different types of criteria and instructions. I now renamed the one JavaDoc that it does not mention `empty criterion`.

**Review by DominikRemo — COMMENTED** [review]
LGTM 👍

Left one small comment that might also help prevent flaky tests in the future.

File: src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java [hunk 2]
```diff
@@ -126,40 +138,61 @@ public List<GradingCriterion> addGradingInstructionsToExercise(Exercise exercise
...
     }
 
+    /**
+     * Accesses the first exercise of a course with the passed type.
+     *
+     * @param course The course which should be searched for the exercise.
+     * @param clazz  The class (type) of the exercise to look for.
+     * @return The first exercise which was found in the course and is of the expected type.
+     */
     public <T extends Exercise> T getFirstExerciseWithType(Course course, Class<T> clazz) {
```
[the remark this answers is not in the document]
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java:149]: Makes sense :)

**Review by DominikRemo — APPROVED** [review]
LGTM, thank you for implementing my feedback

**Review by DominikRemo — APPROVED** [review]

**Pull request merged**